src/pages/VideoDetail/VideoDetail.js

```diff
@@ -2,28 +2,36 @@ import React, { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 import axios from "axios";
 import "./VideoDetail.css";
+
 import { BiLike, BiPlayCircle, BiTime } from "react-icons/bi";
-import { CgTimer } from "react-icons/cg";
-import { AiOutlineEye } from "react-icons/ai";
+import {MdOutlineWatchLater, MdWatchLater} from 'react-icons/md'
+import { AiOutlineEye, AiFillLike } from "react-icons/ai";
+
 import { useAuth } from "context/auth-context/auth-context";
 import { useLike } from "context/like-context/like-context";
+import { useWatchLater } from "context/watch-context/watch-context";
 
 export const VideoDetail = () => {
   const [singleVideoDetail, setSingleVideoDetail] = useState(null);
   const [isLiked, setIsLiked] = useState(false);
+  const [isWatchLater, setIsWatchLater] = useState(false);
 
   const { likedVideos, setLikedVideos } = useLike();
-  const { auth } = useAuth();
+  const { watchLaterVideos, setWatchLaterVideos } = useWatchLater();
 
+  const { auth } = useAuth();
   const { videoId } = useParams();
-
   const { title, views, description, thumbnail, duration, _id, creator } =
     singleVideoDetail ?? {};
 
   useEffect(() => {
     setIsLiked(likedVideos.find((video) => video._id === videoId));
```

Comment:
you can create a utility function to find the video, as this line of code will be repeated at multiple places.